mesh/core/src/main/protobuf/resolver.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package io.linkerd.mesh;
+
+import "dtab.proto";
+
+/**
+ * Supports resolution of a concrete name to a replica set.
+ *
+ * Status: Experimental; this API is not yet finalized.
+ */
+service Resolver {
+
+  rpc GetReplicas (ReplicasReq) returns (Replicas) {}
+  rpc StreamReplicas (ReplicasReq) returns (stream Replicas) {}
+}
+
+message ReplicasReq {
+  /**
+   * A concrete cluster id.
+   */
+  Path id = 1;
+}
+
+/**
+ * Describes a TCP endpoint.
+ */
+message Endpoint {
```

Comment:
"Replica" describes the relationship.  Thing thing is an Endpoint.  Endpoints may be grouped in shards, etc.  (Really _Replicas_ could be called _ReplicatedEndpointSet_ or something if we want to be more verbose).